### PR TITLE
.travis.yml: exercise -std=c89 and address corresponding/related problems.

### PR DESCRIPTION
**Pull request by dot-asm**
WIP is to first see that it actually catches the problem being addressed in #6744.

### CODE REVIEW

**Comment by dot-asm** [comment]
Well, it caught stuff, just not one I was aiming at...

**Comment by dot-asm** [comment]
Now when we know that modified target catches the problem, it's appropriate to show that #6744 actually solves the problem. And on top of that address couple of other nits.

**Comment by dot-asm** [comment]
Red cross from MacOS was relevant. Just in case for reference. What's _GNU/_BSD_SOURCE is about? Without them (and with -ansi) you end up with no fileno, fdopen and some other stuff. So it kind of "merges" C89 language with more recent POSIX API. Now Linux is bailing... Putting back to WIP...

**Comment by dot-asm** [comment]
This is strange. I have no problems compiling with _BSD_SOURCE on my Linux workstation...

**Comment by dot-asm** [comment]
Editorial note. First commit message will be rephrased as "exercise -std=c89 in order to catch corresponding problems."

File: crypto/ui/ui_openssl.c [hunk 1]
```diff
@@ -43,7 +43,7 @@
  * If unistd.h defines _POSIX_VERSION, we conclude that we are on a POSIX
  * system and have sigaction and termios.
  */
-#  if defined(_POSIX_VERSION)
+#  if defined(_POSIX_VERSION) && _POSIX_VERSION>=199309L
```
Comment (richsalz) [inline crypto/ui/ui_openssl.c]: I thought there were issues with short-circuit expressions and defined?

Comment (dot-asm) [reply crypto/ui/ui_openssl.c]: No. There are [warning] issues with e,g. `#if WHATEVER>5` in cases when WHATEVER is not defined. There are real issues with e,g, `#if defined(WHATEVER) && WHATEVER(x,y)>5`, i.e. when WHATEVER is a macro **with parameters**. But there **no** issues with `#if defined(WHATEVER) && WHATEVER>5`.

Comment (richsalz) [reply crypto/ui/ui_openssl.c]: Ah, parameters; thanks.  Reconfirm.

**Comment by slontis** [comment]
Sorry - was away yesterday. Will make sure everything is good on Monday. Thanks for checking so quickly..

**Pull request closed without merging**